### PR TITLE
Add some ore compat/fixes

### PR DESCRIPTION
**Pull request by StockiesLad**
More mod compatibility as well as some adjustments to fix conflicts.

### CODE REVIEW

**Comment by StockiesLad** [comment]
AE2 Compat added.

**Comment by StockiesLad** [comment]
I'm gonna add industrial revolution support and im done (maybe)

**Comment by StockiesLad** [comment]
Ok I can rest now!

**Comment by StockiesLad** [comment]
Updated fabric to prepare for loader plugins

File: Common/src/main/resources/default_configs/betternether.json [hunk 1]
```diff
@@ -42,7 +42,7 @@
       ]
     },
     {
-      "id": "ruby_ore",
+      "id": "nether_ruby_ore",
```
Comment (lukebemish) [inline Common/src/main/resources/default_configs/betternether.json]: Uness you add an `orename` field, this change will break ore dict compat

File: Common/src/main/resources/default_configs/enriched.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      ],
+	  "en_name": "Tanzanite Ore",
+      "types": [
+        "stone"
+      ]
+    },
+	{
+	  "id": "ruby_ore",
+	  "stone": [
+        "deepslate",
```
Comment (lukebemish) [inline Common/src/main/resources/default_configs/enriched.json]: Stones need to be in the same order as the blocks

File: Common/src/main/resources/default_configs/enriched.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      "types": [
+        "stone"
+      ]
+    }
+  ],
+  "provided_ores": [
+    {
+	  "id": "tanzanite_ore",
+	  "stone": [
+        "deepslate",
```
Comment (lukebemish) [inline Common/src/main/resources/default_configs/enriched.json]: Stones need to be in the same order as the blocks

File: Common/src/main/resources/default_configs/enriched.json [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      ],
+	  "en_name": "Ruby Ore",
+      "types": [
+        "stone"
+      ]
+    },
+	{
+	  "id": "sapphire_ore",
+	  "stone": [
+        "deepslate",
```
Comment (lukebemish) [inline Common/src/main/resources/default_configs/enriched.json]: Stones need to be in the same order as the blocks

File: Common/src/main/resources/default_configs/indrev.json [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    {
+      "block_id": [
+        "indrev:lead_ore",
+        "indrev:deepslate_lead_ore"
+      ],
+      "en_name": "Lead Ore",
+      "types": [
+        "stone"
+      ],
+      "id": "sead_ore",
```
Comment (lukebemish) [inline Common/src/main/resources/default_configs/indrev.json]: Might want to fix this

File: Common/src/main/resources/default_configs/modern_industrialization.json [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+      ],
+      "en_name": "Platinum Ore",
+      "types": [
+        "stone"
+      ]
+    },
+	{
+      "id": "tin_ore",
+      "stone": [
+        "stone"
```
Comment (lukebemish) [inline Common/src/main/resources/default_configs/modern_industrialization.json]: You need the same number of stones as block ids

File: Common/src/main/resources/default_configs/modern_industrialization.json [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+      ],
+      "en_name": "Titanium Ore",
+      "types": [
+        "stone"
+      ]
+    },
+	{
+      "id": "tungsten_ore",
+      "stone": [
+        "stone"
```
Comment (lukebemish) [inline Common/src/main/resources/default_configs/modern_industrialization.json]: Same as above

File: changelogs/CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,8 @@
+### Excavated Variants v0.6.3
```
Comment (lukebemish) [inline changelogs/CHANGELOG.md]: Leave this out; I'll write my own changelogs

File: gradle.properties [hunk 9]
```diff
@@ -11,8 +11,8 @@ forge_version=40.1.0
 //forge_ats_enabled=false
 
 # Fabric
-fabric_version=0.51.0+1.18.2
-fabric_loader_version=0.13.3
+fabric_version=0.53.4+1.18.2
```
Comment (lukebemish) [inline gradle.properties]: Any particular reason this is needed?

**Review by lukebemish — CHANGES_REQUESTED** [review]
You might want to go through and fix some stuff up; I have a script that I use to generate these entries for mods with proper tag entries, so I may go through and redo the indrev and MI ones, but otherwise this looks good.

**Comment by lukebemish** [comment]
Also, what are fabric loader plugins and why does the mod need version bumps for this?

**Comment by StockiesLad** [comment]
I'll work on it

**Comment by StockiesLad** [comment]
> Also, what are fabric loader plugins and why does the mod need version bumps for this?

Next version is going to have loader plugins (0.15). I bumped the version to prepare for big changes as I'm sure it'll break things. I undo this if you want.

**Comment by lukebemish** [comment]
Okay, but what are loader plugins?

**Comment by lukebemish** [comment]
Ah, did some research; this shouldn't affect me, no reason to bump it unless they make a breaking change

**Comment by lukebemish** [comment]
Especially since 1.18 support will end fairly soon

**Comment by StockiesLad** [comment]
Alright. I'll reverse it.

**Comment by lukebemish** [comment]
Cool. It won't matter much in 1.19 anyways, since as soon as curseforge gives me an option I'm going to discontinue Fabric support in favor of Quilt.

**Comment by StockiesLad** [comment]
> Cool. It won't matter much in 1.19 anyways, since as soon as curseforge gives me an option I'm going to discontinue Fabric support in favor of Quilt.

Hmmm... that's a bummer. Curseforge supports quilt now FYI. Could I also make a syntax.json so people can just copy it and modify it to add their own support very quickly.

**Comment by lukebemish** [comment]
What do you mean by that?

**Comment by StockiesLad** [comment]
> What do you mean by that?

So I basically have JSON Scripts for ores and stones but without any ID's and with labels to visual demonstrate how it works.

**Comment by StockiesLad** [comment]
Jesus Christ pull requests give me a headache lol.

**Comment by StockiesLad** [comment]
Alright, the scripts should be fixed. I'll probably make a syntax.json and I'll delete it if you don't want it.

**Comment by StockiesLad** [comment]
Ok, I finished that. Now it's up to review.

**Comment by StockiesLad** [comment]
why is it still looking at the old commits lol.

**Comment by StockiesLad** [comment]
Also. I think you should keep fabric even if you aren't going to support it as some others might maintain it for you.

**Comment by lukebemish** [comment]
The project is LGPL and they can feel free to do so themselves, but I will not be supporting fabric

**Comment by lukebemish** [comment]
I'll review it once I'm back at a computer

File: Common/src/main/resources/default_configs/betternether.json (hunk 1 above)
Comment (lukebemish) [reply Common/src/main/resources/default_configs/betternether.json]: Okay, I should specify; either add an `orename` field, or revert this change

File: Common/src/main/resources/default_configs/example.json [hunk 10]
```diff
@@ -0,0 +1,30 @@
+{
+  "mod_id": "", #add mod_id
```
Comment (lukebemish) [inline Common/src/main/resources/default_configs/example.json]: Can you switch these to java-style comments, with a space after the `//`?

File: Common/src/main/resources/default_configs/example.json [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+	  "id": "itemid", #Don't add modid because i will make duplicate ores
+	  "stone": [
+        "deepslate", #add the stone types that this ore already has.
+        "stone"
+      ],
+      "block_id": [
+        "enriched:tanzanite_ore",   #first id is the reference one.
+        "enriched:deepslate_tanzanite_ore" #make sure the block_id[] and stone[] are in the *exact* same order.
+      ],
+	  "en_name": "", #Actual Name, e.g. Ruby Ore
```
Comment (lukebemish) [inline Common/src/main/resources/default_configs/example.json]: It's the english translation name, not so much the "actual name"

File: Common/src/main/resources/default_configs/example.json [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+      "types": [
+        "stone" #stone=overworld, end=end,nether=nether. Add the following for the dimension the blocks will spawn in
+      ]
+    } #add a comma after this bracket to make more than one. But don't have a comma at the last curly bracket in this array.
+  ], #array=provided_ores and provided_stones
+  "provided_ores": [
+    {
+	  "id": "itemid", #Don't add modid because i will make duplicate ores
+	  "stone": [
+        "deepslate", #add the stone types that this ore already has.
```
Comment (lukebemish) [inline Common/src/main/resources/default_configs/example.json]: And yet, they're not in the same order here

File: Common/src/main/resources/default_configs/example.json [hunk 13]
```diff
@@ -0,0 +1,30 @@
+{
+  "mod_id": "", #add mod_id
+  "provided_stones": [
+    {
+	  "id": "modid_itemid", #Don't add ":"
```
Comment (lukebemish) [inline Common/src/main/resources/default_configs/example.json]: That's not what the ID is. The ID is an identifier for the stone. It should be unique. It has nothing to do with the mod/item ID; that's just a convention I follow to guarantee uniqueness

File: Common/src/main/resources/default_configs/example.json [hunk 14]
```diff
@@ -0,0 +1,30 @@
+{
+  "mod_id": "", #add mod_id
+  "provided_stones": [
+    {
+	  "id": "modid_itemid", #Don't add ":"
+	  "block_id": "modid:itemid", #e.g. enriched:dark_granite
+	  "en_name": "", #Actual name, e.g. "Dark Granite
```
Comment (lukebemish) [inline Common/src/main/resources/default_configs/example.json]: Close quote here

File: Common/src/main/resources/default_configs/example.json [hunk 15]
```diff
@@ -0,0 +1,30 @@
+{
+  "mod_id": "", #add mod_id
+  "provided_stones": [
+    {
+	  "id": "modid_itemid", #Don't add ":"
+	  "block_id": "modid:itemid", #e.g. enriched:dark_granite
+	  "en_name": "", #Actual name, e.g. "Dark Granite
+      "types": [
+        "stone" #stone=overworld, end=end,nether=nether. Add the following for the dimension the blocks will spawn in
```
Comment (lukebemish) [inline Common/src/main/resources/default_configs/example.json]: It has nothing to do with the dimension; it is a category for the stone that associates ores with stones.

File: gradle.properties [hunk 16]
```diff
@@ -11,7 +11,7 @@ forge_version=40.1.0
 //forge_ats_enabled=false
 
 # Fabric
-fabric_version=0.51.0+1.18.2
+fabric_version=0.51.1+1.18.2
```
Comment (lukebemish) [inline gradle.properties]: Might want to revert this change; no reason to break backwards compat if I don't need to

**Review by lukebemish — CHANGES_REQUESTED** [review]

**Comment by StockiesLad** [comment]
Alright that should do it.

File: Common/src/main/resources/default_configs/example.json (hunk 13 above)
Comment (lukebemish) [reply Common/src/main/resources/default_configs/example.json]: This was meant for the ID, not the modid

**Comment by StockiesLad** [comment]
I was so close to +420 lines lmao

**Pull request merged**